### PR TITLE
Refresh vault token

### PR DESCRIPTION
**Pull request by habuma**

### CODE REVIEW

File: spring-cloud-services-spring-connector/src/main/java/io/pivotal/spring/cloud/service/config/VaultTokenRenewer.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * The renew rate can be configured by setting `vault.token.renew.rate` to some value that is the 
+ * renewal rate in milliseconds. The renewal time-to-live can be specified with by setting
+ * `vault.token.ttl` to some value indicating the time-to-live in seconds.
+ * 
+ * @author cwalls
+ */
+@Configuration
+@ConditionalOnBean(ConfigClientProperties.class)
+@ConditionalOnProperty(name="spring.cloud.config.token")
+public class VaultTokenRenewer {
```
Comment (ojhughes) [inline spring-cloud-services-spring-connector/src/main/java/io/pivotal/spring/cloud/service/config/VaultTokenRenewer.java]: Pedantic comment but `VaultTokenRenewerAutoConfiguration` would mirror the naming of the other auto config classes

Comment (habuma) [inline spring-cloud-services-spring-connector/src/main/java/io/pivotal/spring/cloud/service/config/VaultTokenRenewer.java]: Your point is well taken, but...I went back and forth on that for several hours yesterday. Technically it's only a configuration class because it's what ends up in `spring.factories` under `org.springframework.boot.autoconfigure.EnableAutoConfiguration`. Ideally I wanted to extract the guts of it into a component class called `VaultTokenRenewer`, but when I did that refactoring, there wasn't anything left in the configuration class.

In short, this is a component class (maybe had I annotated it with `@Component` that would've been more obvious--I'm not sure if that would've even worked, though). But it's treated like a `@Configuration` class for the sake of auto-configuration. 

But yeah...you're only asking for a class name change. I suppose that if this moves things forward then I can do that.
